src/modules/colorPicker/ColorPickerUI/Keyboard/KeyboardMonitor.cs

```diff
@@ -72,8 +72,16 @@ private void Hook_KeyboardPressed(object sender, GlobalKeyboardHookEventArgs e)
             // ESC pressed
             if (virtualCode == KeyInterop.VirtualKeyFromKey(Key.Escape))
             {
-                _appStateHandler.HideColorPicker();
-                PowerToysTelemetry.Log.WriteEvent(new ColorPickerCancelledEvent());
+                if (_appStateHandler.IsColorPickerEditorVisible())
+                {
+                    _appStateHandler.HideColorPickerEditor();
+                }
+                else
+                {
+                    _appStateHandler.HideColorPicker();
+                    PowerToysTelemetry.Log.WriteEvent(new ColorPickerCancelledEvent());
+                }
```

Comment:
what about zoom window?

Reply:
If you press esc with the zoom window, the whole colour picker closes which is the same as the current version. Does this need changing?

Reply:
you are right, it will fire close event and zoom window will close itself as well, I don't remember my own code :)

Reply:
Great - thanks for testing it! 😀

Reply:
Thanks for the contribution!